examples/README-authz.md

```diff
@@ -0,0 +1,353 @@
+
+Let's try produce some messages to topic 'a_messages' as `team-b-client`:
+
+```
+bin/kafka-console-producer.sh --broker-list kafka:9092 --topic a_messages --producer.config ~/team-b-client.properties
+Message 1
+```
+
+We get `Not authorized to access topics: [a_messages]` error as we expected. Let's try to produce to topic `b_messages`:
+
+```
+bin/kafka-console-producer.sh --broker-list kafka:9092 --topic b_messages --producer.config ~/team-b-client.properties
+Message 1
+Message 2
+Message 3
+```
+
+This should work fine.
+
+What about producing to topic `x_messages`. `team-b-client` is only supposed to be able to read from such a topic.
+
+```
+bin/kafka-console-producer.sh --broker-list kafka:9092 --topic x_messages --producer.config ~/team-b-client.properties
+Message 1
+```
+
+We get a `Not authorized to access topics: [x_messages]` error as we expected. 
+Client `team-a-client`, on the other hand, should be able to write to such a topic:
+
+```
+bin/kafka-console-producer.sh --broker-list kafka:9092 --topic x_messages --producer.config ~/team-a-client.properties
+Message 1
+```
+
+However, we again receive `Not authorized to access topics: [x_messages]`. What's going on?
+The reason for failure is that while `team-a-client` can write to `x_messages` topic, it does not have a permission to create one if it does not yet exist.
+
+We now need a power user that can create a topic with all the proper settings - like the right number of partitions and replicas.
+
+Let's create a configuration for user `bob` who has full ability to manage everything on Kafka cluster `cluster2`.
+
+First, `bob` will authenticate to Keycloak server with username and password and get a refresh token.
+
+```
+export TOKEN_ENDPOINT=http://keycloak:8080/auth/realms/kafka-authz/protocol/openid-connect/token
+REFRESH_TOKEN=$(./oauth.sh -q bob)
+```
+
+This will prompt you for a password. Type 'bob-password'.
+
+We can inspect the refresh token:
+
+    ./jwt.sh $REFRESH_TOKEN
+
+By default this is a long-lived refresh token that does not expire.
+
+Now we will create the configuration file for `bob`:
+
+```
+cat > ~/bob.properties << EOF
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=OAUTHBEARER
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  oauth.refresh.token="$REFRESH_TOKEN" \
+  oauth.client.id="kafka-cli" \
+  oauth.token.endpoint.uri="http://keycloak:8080/auth/realms/kafka-authz/protocol/openid-connect/token" ;
+sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler
+EOF
+```
+
+Note that we use `kafka-cli` public client for clientId. 
+Since that is a public client it does not require any secret.
+We can use this, because we authenticate with a token directly (in this case a refresh token is used to request an access token behind the scenes which is then sent to Kafka broker for authentication).
```

Comment:
```suggestion
We can use this because we authenticate with a token directly (in this case a refresh token is used to request an access token behind the scenes which is then sent to Kafka broker for authentication, and we already did the authentication for the refresh token).
```